cfg/builder/builder_walk.cc

```diff
@@ -621,11 +663,17 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::ExpressionPtr &what, BasicBlo
                     unconditionalJump(current, headerBlock, cctx.inWhat, s.loc);
 
                     LocalRef blockrv = cctx.newTemporary(core::Names::blockReturnTemp());
-                    auto blockLast = walk(cctx.withTarget(blockrv)
-                                              .withBlockBreakTarget(cctx.target)
-                                              .withLoopScope(headerBlock, postBlock, true)
-                                              .withSendAndBlockLink(link),
-                                          s.block()->body, argBlock);
+                    BasicBlock *blockLast;
+                    {
+                        auto newCctx = cctx.withTarget(blockrv)
+                                           .withBlockBreakTarget(cctx.target)
+                                           .withLoopScope(headerBlock, postBlock, true)
+                                           .withSendAndBlockLink(link);
+                        if (isKernelLambda(s)) {
+                            newCctx.isInsideLambda = true;
+                        }
```

Comment:
Why did you change it from `newCctx.isInsideLambda = isKernelLambda(s);` to this?

Reply:
It doesn’t reset. If you’re in an each block inside a lambda, return returns from the lambda, not the method